pkg/datasources/streamlits_acceptance_test.go

```diff
@@ -0,0 +1,189 @@
+package datasources_test
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"regexp"
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/snowflakeroles"
+	"github.com/stretchr/testify/require"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestAcc_Streamlits(t *testing.T) {
+	databaseId := acc.TestClient().Ids.DatabaseId()
+	schemaId := acc.TestClient().Ids.SchemaId()
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schemaId)
+	ctx := context.Background()
+	acc.TestAccPreCheck(t)
+	stage, stageCleanup := acc.TestClient().Stage.CreateStageInSchema(t, schemaId)
+	t.Cleanup(stageCleanup)
+	err := acc.Client(t).Sessions.UseSchema(ctx, schemaId)
+	require.NoError(t, err)
+	// warehouse is needed because default warehouse uses lowercase, and it fails in snowflake.
+	warehouse, warehouseCleanup := acc.TestClient().Warehouse.CreateWarehouse(t)
+	t.Cleanup(warehouseCleanup)
+	integrationId := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	networkRule, networkRuleCleanup := acc.TestClient().NetworkRule.CreateNetworkRule(t)
+	t.Cleanup(networkRuleCleanup)
+	_, err = acc.Client(t).ExecForTests(ctx, fmt.Sprintf(`CREATE EXTERNAL ACCESS INTEGRATION %s ALLOWED_NETWORK_RULES = (%s) ENABLED = TRUE`, integrationId.Name(), networkRule.ID().Name()))
```

Comment:
maybe better to extract it into TestClient rather than copy paste in multiple tests

Reply:
Done. I wasn't sure if we want to do this for resources missing from SDK.